core/src/main/scala/org/apache/spark/api/python/WriteInputFormatTestDataGenerator.scala

```diff
@@ -104,6 +105,7 @@ private[python] class WritableToDoubleArrayConverter extends Converter[Any, Arra
  * This object contains method to generate SequenceFile test data and write it to a
  * given directory (probably a temp directory)
  */
+@deprecated("This class will be move to `test`.", "3.5.2")
```

Comment:
Hmmmm ... things here aren't really an API....

Reply:
@HyukjinKwon . I understand your point. However, this is **exposed** since Apache Spark 1.1.0. We don't know if there is someone take advantage of this API or not. If we say simply `it's a mistake` and remove **exposed** methods (Apache Spark 1.1.0 ~ 3.5.x) without deprecation, it would be a violation of Semantic Versioning. We have `MIMA` check to prevent those kind of weird situation. 
> My concern is that we're marking deprecation, and it sounds like we're admitting those are actually an API.

Reply:
To me, the package name `org/apache/spark/api/python` itself clearly implies this was added as a part of API at that time.